### PR TITLE
[netcdf-c] Update to 4.7.3 and switched to use targets to generate valid targets

### PR DESCRIPTION
**Pull request by Neumann-A**
**Describe the pull request**

- netcdf-c is not correctly configuring due to missing includes and library dependencies (zlib/szip)
- the generated target was not linking correctly to libcurl (Link interface included both debug and optimized version of libcurl without a generator expressions. Reason for that was a remove duplicates call in the cmakelists.txt which removed duplicated optimized/debug keywords)
- ~~the targets were installed into the wrong directory so that find_package was unable to find netcdf (old directory was netcdf-c which does not work if the usage is find_package(netCDF) ) The targets/config are also called netCDF and not netCDF-c.~~ Checked CMake docs again and it searches in `name*` so its ok

### CODE REVIEW

**Review by JackBoosY — COMMENTED** [review]
Can we move find dependencies from `portfile.cmake` to `CMakeLists.txt`?This will make `portfile.cmake` more concise.

**Comment by PhoebeHui** [comment]
@Neumann-A , thanks for the PR!

I double checked, It involved in regressions when 'set(VCPKG_BUILD_TYPE release)', the failures like:
CMake Error at scripts/cmake/vcpkg_prettify_command.cmake:19 (if):
  if given arguments:

    "-DSZIP_LIBRARY:STRING=debug" "SZIP_DEBUG-NOTFOUND" "optimized" "F:/VCPKG/numan/vcpkg/installed/x64-windows/lib/libszip.lib" "MATCHES" "( )"

  Unknown arguments specified
Call Stack (most recent call first):
  scripts/cmake/vcpkg_execute_required_process.cmake:71 (vcpkg_prettify_command)
  scripts/cmake/vcpkg_configure_cmake.cmake:312 (vcpkg_execute_required_process)
  ports/netcdf-c/portfile.cmake:35 (vcpkg_configure_cmake)
  scripts/ports.cmake:94 (include)

**Comment by Neumann-A** [comment]
So I updated the library and switched everything to use targets instead of library names. So it should link correctly if the other targets are correct. Lets see what CI tells us

**Comment by JackBoosY** [comment]
You fixed the static build but broke the dynamic build : )

File: ports/netcdf-c/portfile.cmake [hunk 1]
```diff
@@ -1,34 +1,36 @@
...
 else()
     set(NC_USE_STATIC_CRT OFF)
 endif()
+#NC_EXTRA_DEPS 
+find_library(ZLIB_RELEASE NAMES z zlib PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
+find_library(ZLIB_DEBUG NAMES z zlib zd zlibd PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+find_library(SZIP_RELEASE NAMES libsz libszip szip sz PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
+find_library(SZIP_DEBUG NAMES libsz libszip szip sz libsz_D libszip_D szip_D sz_D szip_debug PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
+#find_library(CURL_RELEASE NAMES curl libcurl PATHS "${CURRENT_INSTALLED_DIR}/lib" NO_DEFAULT_PATH)
+#find_library(CURL_DEBUG NAMES curl_d libcurl_d curld libcurld curl libcurl PATHS "${CURRENT_INSTALLED_DIR}/debug/lib" NO_DEFAULT_PATH)
```
Comment (ras0219-msft) [inline ports/netcdf-c/portfile.cmake]: These don't currently appear to be used (the cmake build doesn't inherit variables from outside unless explicitly passed in)

File: ports/netcdf-c/use_targets.patch [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+-    IF(NC_FIND_SHARED_LIBS)
+-      SET(NC_HDF5_LINK_TYPE "shared")
+-      SET(NC_HDF5_LINK_TYPE_UPPER "SHARED")
+-      ADD_DEFINITIONS(-DH5_BUILT_AS_DYNAMIC_LIB)
+-    ELSE(NC_FIND_SHARED_LIBS)
+-      SET(NC_HDF5_LINK_TYPE "static")
+-      SET(NC_HDF5_LINK_TYPE_UPPER "STATIC")
+-      ADD_DEFINITIONS(-DH5_BUILT_AS_STATIC_LIB)
+-    ENDIF(NC_FIND_SHARED_LIBS)
++    #IF(NC_FIND_SHARED_LIBS)
```
Comment (JackBoosY) [inline ports/netcdf-c/use_targets.patch]: Can you delete these commented codes directly to reduce the patch size?

Thanks.

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/netcdf-c/use_targets.patch (hunk 2 above)
Comment (Neumann-A) [reply ports/netcdf-c/use_targets.patch]: if i did not miss some spot https://github.com/microsoft/vcpkg/pull/9361/commits/f30f89a98c2ec2b2d6781dca24506602ee179805 should fullfil this request

**Review by JackBoosY — APPROVED** [review]

**Review by PhoebeHui — APPROVED** [review]

**Pull request merged**